services/app-web/src/hooks/useFileUpload.ts

```diff
@@ -0,0 +1,89 @@
+import { useEffect, useState } from "react"
+import { FileItemT } from "../components"
+
+// Intended for use with FileUpload component 
+const useFileUpload = (shouldValidate = false) => {
+    const [fileItems, setFileItems] = useState<FileItemT[]>([])
+    const [hasValidFiles, setHasValidFiles] = useState(false)
+    const [hasLoadingFiles, setHasLoadingFiles] = useState(false)
+
+    const onFileItemsUpdate = async ({
+        fileItems,
+    }: {
+        fileItems: FileItemT[]
+    }) => {
+        setFileItems((prevValue) =>
+            fileItems.length !== prevValue.length ||
+            !prevValue.every((el) => fileItems.includes(el))
+                ? fileItems
+                : prevValue
+        )
+    }
+    useEffect(() => {
+        const allFilesComplete = fileItems.every(
+            (item) => item.status === 'UPLOAD_COMPLETE'
+        )
+        setHasValidFiles(allFilesComplete)
+
+        if (!allFilesComplete) {
+            const updatedLoading = fileItems.some(
+                (item) =>
+                    item.status === 'PENDING' || item.status === 'SCANNING'
+            )
+            setHasLoadingFiles(updatedLoading)
+        }
+    }, [fileItems])
+
+    const hasNoFiles = fileItems.length === 0;
+
+    const fileUploadErrorMessage = shouldValidate? hasLoadingFiles
```

Comment:
I know we do these nested ternaries in a few places.  This one took me a while.  I'll leave it up to the team whether this or something like the following is easier to follow.
```javascript
let fileUploadErrorMessage = undefined;

if (shouldValidate) {
  if (hasLoadingFiles) {
    fileUploadErrorMessage = 'You must wait for all documents to finish uploading before continuing';
  } else if (fileItems.length === 0) {
    fileUploadErrorMessage = 'You must upload at least one document';
  } else if (!hasValidFiles) {
    fileUploadErrorMessage = 'You must remove all documents with error messages before continuing';
  }
}
```

Reply:
yeah I agree, especially because this one has multiple `undefined` cases now.   let me just make that change.